common/faceting.js

```diff
@@ -641,16 +760,158 @@
                         return defer.promise;
                     };
 
+                    // Zoom the set into the middle 50% of the buckets
+                    scope.zoomInPlot = function () {
+                        try {
+                            // we don't want the user to keep zooming in infinitely to save memory space
+                            if (scope.histogramDataStack.length >= 20) { throw new Error("Maximum data stack size reached"); }
+
+                            // If the range between min/max is less than the number of buckets, no reason to zoom more
+                            // the bars will only get bigger without any more precision being offered
+                            if (isColumnOfType("int")) {
+                                if (scope.rangeOptions.absMax - scope.rangeOptions.absMin <= numBuckets) {
+                                    throw new Error("Can't zoom anymore");
```

Comment:
Should we show these errors as alert to the users?

Reply:
Same comment as above, this case _should_ never be reached.